onnx/version_converter/convert.h

```diff
@@ -0,0 +1,162 @@
+// Default converter for ONNX models between different opset versions
+// in the default domain ("" or "ai.onnx").
+
+#pragma once
+
+#include "onnx/version_converter/BaseConverter.h"
+#include "onnx/version_converter/adapters/NoPreviousVersionAdapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+class DefaultVersionConverter : public BaseVersionConverter {
+  private:
+    bool DEBUG = false;
+
+    bool searchOpDomainMap(const std::unordered_map<std::string, std::map<
+      int64_t, const OpSchema*>>& op_domain_map, int64_t curr_version) const {
+      const auto& version_it = op_domain_map.find("");
+      return version_it != op_domain_map.end() &&
+          version_it->second.find(curr_version) !=
+          version_it->second.end();
+    }
+
+  public:
+    DefaultVersionConverter() {
+      // Register adapters to the version converter
+      registerAdapter(std::unique_ptr<Adapter>(new NoPreviousVersionAdapter("Cos",
```

Comment:
use `make_unique` instead, never directly call `new` or `delete`.

Reply:
Doesn't compile due to Adapter being an abstract class.  @houseroad?

Reply:
need to change this for adapters, too